src/passes/PrintCallGraph.cpp

```diff
@@ -34,72 +35,133 @@ struct PrintCallGraph : public Pass {
   bool modifiesBinaryenIR() override { return false; }
 
   void run(Module* module) override {
-    std::ostream& o = std::cout;
-    o << "digraph call {\n"
-         "  rankdir = LR;\n"
-         "  subgraph cluster_key {\n"
-         "    node [shape=box, fontname=courier, fontsize=10];\n"
-         "    edge [fontname=courier, fontsize=10];\n"
-         "    label = \"Key\";\n"
-         "    \"Import\" [style=\"filled\", fillcolor=\"turquoise\"];\n"
-         "    \"Export\" [style=\"filled\", fillcolor=\"gray\"];\n"
-         "    \"Indirect Target\" [style=\"filled, rounded\", "
-         "fillcolor=\"white\"];\n"
-         "    \"A\" -> \"B\" [style=\"filled, rounded\", label = \"Direct "
-         "Call\"];\n"
-         "  }\n\n"
-         "  node [shape=box, fontname=courier, fontsize=10];\n";
-
-    // Defined functions
-    ModuleUtils::iterDefinedFunctions(*module, [&](Function* curr) {
-      std::cout << "  \"" << curr->name
-                << "\" [style=\"filled\", fillcolor=\"white\"];\n";
-    });
-
-    // Imported functions
-    ModuleUtils::iterImportedFunctions(*module, [&](Function* curr) {
-      o << "  \"" << curr->name
-        << "\" [style=\"filled\", fillcolor=\"turquoise\"];\n";
-    });
-
-    // Exports
-    for (auto& curr : module->exports) {
-      if (curr->kind == ExternalKind::Function) {
-        Function* func = module->getFunction(curr->value);
-        o << "  \"" << func->name
-          << "\" [style=\"filled\", fillcolor=\"gray\"];\n";
-      }
-    }
+    Name caller = getPassOptions().getArgumentOrDefault(
+      "func",
+      "");
 
-    struct CallPrinter : public PostWalker<CallPrinter> {
+    struct GraphCreator : public PostWalker<GraphCreator> {
       Module* module;
       Function* currFunction;
       std::set<Name> visitedTargets; // Used to avoid printing duplicate edges.
       std::vector<Function*> allIndirectTargets;
-      CallPrinter(Module* module) : module(module) {
+      std::multimap<Name, Name> full_graph; // First param is the caller/parent, second is the callee/child
```

Comment:
```suggestion
      std::multimap<Name, Name> fullGraph; // First param is the caller/parent, second is the callee/child
```

Reply:
Changed in https://github.com/WebAssembly/binaryen/pull/6354/commits/10f8643e345aa8176ecbe97b00e8c62c0dc09d73